Xamarin.Essentials/Calendar/Calendar.android.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Android.Content;
+using Android.Database;
+using Android.Provider;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Calendar
+    {
+        const string andCondition = "AND";
+
+        static bool PlatformIsSupported => true;
+
+        static async Task PlatformRequestCalendarReadAccess() => await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+        static async Task PlatformRequestCalendarWriteAccess() => await Permissions.RequireAsync(PermissionType.CalendarWrite);
+
+        static async Task<List<DeviceCalendar>> PlatformGetCalendarsAsync()
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            var calendarsUri = CalendarContract.Calendars.ContentUri;
+            var calendarsProjection = new List<string>
+            {
+                CalendarContract.Calendars.InterfaceConsts.Id,
+                CalendarContract.Calendars.InterfaceConsts.CalendarDisplayName,
+                CalendarContract.Calendars.InterfaceConsts.CalendarAccessLevel
+            };
+            var queryConditions = $"{CalendarContract.Calendars.InterfaceConsts.Deleted} != 1";
+
+            var cur = Platform.AppContext.ApplicationContext.ContentResolver.Query(calendarsUri, calendarsProjection.ToArray(), queryConditions, null, null);
```

Comment:
does  "cur" implement IDisposable, if so, we should use a using statement here

Reply:
Altered to match GetEventById

Reply:
This one still needs a using statement